quantum/split_common/split_util.c

```diff
@@ -135,6 +132,9 @@ __attribute__((weak)) bool is_keyboard_master(void) {
 void split_pre_init(void) {
     isLeftHand = is_keyboard_left();
```

Comment:
This will infinite loop on MASTER_LEFT/MASTER_RIGHT as the slave transport is not established.

Reply:
Can you explain? You mean, it will not return as long as there is no slave that answers the master's messages?

Reply:
> I just tested with my UART split board and the master does not stay in that loop after connecting USB. It works without a slave as well.

This is a board that uses EE_HANDS/SPLIT_HAND_PIN for handedness?

Reply:
Yes, it is.

Reply:
in that case `is_keyboard_left` calls `is_keyboard_master` before transport_slave_init happens. Thus it will never get its "slave" assignment as there will never be anything listening.

Reply:
I moved the slave initialization forward in a later commit: https://github.com/qmk/qmk_firmware/pull/9742/files#diff-57e6bd7db0b3e56c607994b241d7b945R132-R134